src/flow/maximum_flow.jl

```diff
@@ -0,0 +1,141 @@
+"""
+Abstract type that allows users to pass in their preferred Algorithm
+"""
+abstract AbstractFlowAlgorithm
+
+"""
+Forces the maximum_flow function to use Edmonds Karp\'s maximum flow algorithm.
+"""
+type EdmondsKarpAlgorithm <: AbstractFlowAlgorithm
+end
+
+"""
+Forces the maximum_flow function to use Dinic\'s maximum flow algorithm.
+"""
+type DinicAlgorithm <: AbstractFlowAlgorithm
+end
+
+"""
+Forces the maximum_flow function to use the Push-Relabel maximum flow algorithm.
+"""
+type PushRelabelAlgorithm <: AbstractFlowAlgorithm
+end
+
+"""
+Type that returns 1 if a forward edge exists, and 0 otherwise
+"""
+
+type DefaultCapacity <: AbstractArray{Int, 2}
+    flow_graph::LightGraphs.DiGraph
+    nv::Int
+    DefaultCapacity(flow_graph::LightGraphs.DiGraph) = new(flow_graph, nv(flow_graph))
+end
+
+getindex(d::DefaultCapacity, s::Int, t::Int) = if has_edge(d.flow_graph, s , t) 1 else 0 end
+size(d::DefaultCapacity) = (d.nv, d.nv)
+transpose(d::DefaultDistance) = d
```

Comment:
Oops. I'll change that. But considering that DefaultCapacity isn't symmetric, the transpose may not be the same matrix. Is there any function to reverse the graph (reverse all the edges?)